engine/consensus/matching/engine.go

```diff
@@ -260,20 +254,9 @@ func (e *Engine) onReceipt(originID flow.Identifier, receipt *flow.ExecutionRece
 	// eventually request the parent receipt with the `requestPending` function
 	err = e.receiptValidator.Validate(receipt)
 	if err != nil {
-		return fmt.Errorf("failed to process execution receipt: %w", err)
-	}
-
-	// add the receipt to the mempool so that the builder can incorporate it in
-	// a block payload
-	added := e.receipts.Add(receipt)
```

Comment:
seems `receipts` is not being used anywhere, so remove it for now along with its metrics